src/IIIFPresentation/Repository/Paths/PathParser.cs

```diff
@@ -25,5 +25,31 @@ public static AssetId GetAssetIdFromNamedQueryCanvasId(this Canvas canvas, ILogg
             throw new FormatException($"Unable to extract AssetId from {canvas.Id}");
         }
     }
+    
+    public static string GetCanvasId(Models.API.Manifest.CanvasPainting? canvasPainting, int customerId)
+    {
+        var canvasId = canvasPainting?.CanvasId.ThrowIfNull(nameof(canvasPainting));
```

Comment:
Looking at this, param should be `Models.API.Manifest.CanvasPainting canvasPainting`? Currently `Models.API.Manifest.CanvasPainting? canvasPainting` makes it look like it can accept a null `CanvasPainting` but that would result in an error being thrown